ibis/backends/pandas/execution/generic.py

```diff
@@ -432,13 +433,17 @@ def execute_aggregation_dataframe(
 
     scope = scope.merge_scope(Scope({op.table.op(): source}, timecontext))
 
-    pieces = [
-        pd.Series(
-            execute(metric, scope=scope, timecontext=timecontext, **kwargs),
-            name=metric.get_name(),
+    pieces = []
+    for metric in op.metrics:
+        result = execute(
+            metric, scope=scope, timecontext=timecontext, **kwargs
         )
-        for metric in op.metrics
-    ]
+
+        if isinstance(metric, ir.DestructValue):
```

Comment:
hmm can you make a single function to do this (e.g. you can still call coerce_to_dataframe), but conceptually this should be something like

```
pieces = [ coerce_to_output_type(metric, execute(metric, scope=scope, timecontext=timecontext, **kwargs)) for metric in op.metrics]
```

Reply:
Good point. I made a new function `coerce_to_output` for this.